src/ast/async_event_types.h

```diff
@@ -59,5 +59,19 @@ struct Time
   }
 } __attribute__((packed));
 
+struct Buf
+{
+  uint8_t length;
+  uint8_t* content;
+
+  std::vector<llvm::Type*> asLLVMType(ast::IRBuilderBPF& b, size_t length)
+  {
+    return {
+      b.getInt8Ty(),                               // buffer length
+      llvm::ArrayType::get(b.getInt8Ty(), length), // buffer content
+    };
+  }
+};
```

Comment:
So the compiler doesn't insert padding. I believe it'll be a noop on all the architectures we support but it's better to document it anyways.
```suggestion
} __attribute__((packed));
```